src/redux/shoppingCart.js

```diff
@@ -0,0 +1,48 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialCartState = {
+  items: {},
+  subtotal: 0,
+  tax: 0.15,
+  totalQuantity: 0,
+  totalAmount: 0,
+};
+
+const shoppingCartSlice = createSlice({
+  name: 'shoppingCart',
+  initialState: initialCartState,
+  reducers: {
+    addItem: (state, action) => {
+      const { album, quantity } = action.payload;
+      const existingItem = state.items[album.id];
+
+      if (existingItem) {
+        // If the item already exists in the cart, increase the quantity
+        existingItem.quantity += quantity;
+      } else {
+        // Otherwise, add a new item
+        state.items[album.id] = {
+          album,
+          quantity,
+        };
+      }
+
+      // Update the other cart fields
+      state.subtotal += album.price * quantity;
+      state.totalQuantity += quantity;
+      state.totalAmount += album.price * quantity;
```

Comment:
total should take into account tax as well: (eg: `state.totalAmount += ((album.price * quantity) * (1+ state.tax))` or `state.totalAmount = state.subtotal + (state.subtotal * state.tax)`